site/source/docs/building_from_source/index.rst

```diff
@@ -11,23 +11,28 @@ Emscripten's core codebase, which is in the main "emscripten" repo, does not nee
 Building LLVM
 -------------
 
-For using the LLVM wasm backend (recommended), simply build normal upstream LLVM from the monorepo, including clang and wasm-ld (using something like ``-DLLVM_ENABLE_PROJECTS=lld;clang'``) and the wasm backend (which is included by default; just don't disable it), following `that project's instructions <http://llvm.org/docs/CMake.html>`_. For example, something like this can work:
+For using the LLVM wasm backend (recommended), simply build normal upstream LLVM from the `monorepo <https://github.com/llvm/llvm-project>`_, including clang and wasm-ld (using something like ``-DLLVM_ENABLE_PROJECTS='lld;clang'``) and the wasm backend (which is included by default; just don't disable it), following `that project's instructions <http://llvm.org/docs/CMake.html>`_. For example, something like this can work:
 
   ::
 
       mkdir build
-      cd build
-      cmake .. -DCMAKE_BUILD_TYPE=Release -DLLVM_ENABLE_PROJECTS='lld;clang' -DLLVM_TARGETS_TO_BUILD="host;WebAssembly" -DLLVM_INCLUDE_EXAMPLES=OFF -DLLVM_INCLUDE_TESTS=OFF
+      cd build/
+      cmake ../llvm -DCMAKE_BUILD_TYPE=Release -DLLVM_ENABLE_PROJECTS='lld;clang' -DLLVM_TARGETS_TO_BUILD="host;WebAssembly" -DLLVM_INCLUDE_EXAMPLES=OFF -DLLVM_INCLUDE_TESTS=OFF
+      make -j$(nproc)  # no need to install
 
 Please refer to the upstream docs for more detail.
 
+The version to select for e.g. 1.38.43 is --- TODO ---.
```

Comment:
I don't think you need this.  Users building from source normally just want ToT.

Reply:
Hmm, in my experience I needed 1 patch (https://github.com/kripken/emscripten-fastcomp/pull/195) for fastcomp but otherwise stuck to identified, reproducible releases. I'll add kripken's link as well.

Reply:
I guess it depends who you are targeting with these instructions.  If you want to build a pinned release from source then `emsdk` is the easiest way to do that.  It can download and build everything from source in a single command.    I assume these instructions are targetting people who want to work on the sources in place?   

However I don't see any harm in documenting that the fastcomp and binaryen are  tagged with emscripten releases.  Remember upstream llvm has no such tags though.

Reply:
If emsdk can rebuild everything from source, including a patch or two, then that sounds like something to write proeminently here! I had no idea.
Is that working for -upstream yet, I didn't find the right 'emsdk install' tag?

Reply:
emsdk only knows how to build pinned releases.     So you don't get a chance to edit or patch the code in that case.

Thats why I assumed these instructions were more directed at people who want to work on the codebase. 

Perhaps we should add link to this doc saying something like: `Unless you need to modify of customize llvm or binaryen then emsdk is the easiest way to build them from srouce.  See <link>.`?

Reply:
I only found https://github.com/emscripten-core/emsdk/blob/master/README.md#how-do-i-track-the-latest-emscripten-development-with-the-sdk which is about rebuilding master/incoming for fastcomp, maybe there's a better link.